client/components/localized-moment/test/index.js

```diff
@@ -71,61 +60,70 @@ describe.each( [
 	it( 'renders localized date on a repeated language switch', async () => {
 		const store = createStore( reducer, createState( 'en' ) );
 
-		const wrapper = mount(
+		const { container } = render(
 			<MomentProvider store={ store }>
 				<LocalizedLabel date="2018-11-01" />
 			</MomentProvider>
 		);
 
-		expect( wrapper.text() ).toEqual( enLabel );
+		expect( container.firstChild ).toHaveTextContent( enLabel );
```

Comment:
We could use screen queries here if we want:

```suggestion
		expect( screen.getByText( enLabel ) ).toBeVisible();
```

This is as recommended by the Kent C. Dodds [here](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen) and is applicable for the rest of the PR.